src/activity_actor.cpp

```diff
@@ -2525,6 +2524,95 @@ std::unique_ptr<activity_actor> milk_activity_actor::deserialize( JsonIn &jsin )
     return actor.clone();
 }
 
+static bool check_if_disassemble_okay( item_location target, Character &who )
+{
+    item *disassembly = target.get_item();
+
+    // item_location::get_item() will return nullptr if the item is lost
+    if( !disassembly ) {
+        who.add_msg_player_or_npc(
+            _( "You no longer have the in progress disassembly in your possession.  "
+               "You stop disassembling.  "
+               "Reactivate the in progress disassembly to continue disassembling." ),
+            _( "<npcname> no longer has the in progress disassembly in their possession.  "
+               "<npcname> stops disassembling." ) );
+        return false;
+    }
+    return true;
+}
+
+void disassemble_activity_actor::start( player_activity &act, Character &who )
+{
+    if( act.targets.empty() ) {
+        act.set_to_null();
+        return;
+    }
+
+    if( act.targets.back()->typeId() != itype_disassembly ) {
+        target = who.create_in_progress_disassembly( act.targets.back() );
+    } else {
+        target = act.targets.back();
+    }
+    act.targets.pop_back();
+
+    if( !check_if_disassemble_okay( target, who ) ) {
+        act.set_to_null();
+        return;
+    }
+
+    // We already checked if this is nullptr above
+    item *craft = target.get_item();
+    double counter = craft->item_counter;
+
+    act.moves_left = moves_total - ( counter / 10'000'000.0 ) * moves_total;
```

Comment:
The "quote marks" above are, umm, rather confusing.

Reply:
Its a convention used by the existing code for the usage of item counters as progress and in that context does make some sense, though I agree at first glance its weird.